lib/projects/localDev/LocalDevUIWebsocketServer.ts

```diff
@@ -0,0 +1,115 @@
+import { WebSocketServer, WebSocket } from 'ws';
+import {
+  isPortManagerServerRunning,
+  requestPorts,
+} from '@hubspot/local-dev-lib/portManager';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { LOCAL_DEV_UI_WEBSOCKET_MESSAGE_TYPES } from '../../constants';
+import { LocalDevUIWebsocketMessage } from '../../../types/LocalDev';
+import { ProjectConfig } from '../../../types/Projects';
+const SERVER_INSTANCE_ID = 'local-dev-ui-websocket-server';
+
+const LOG_PREFIX = '[LocalDevUIWebsocketServer] ';
+
+class LocalDevUIWebsocketServer {
```

Comment:
This is still WIP - started working on it before I realized that the rest of the work in this PR was necessary. But figured it made sense to keep it in here and add to it in following PRs